### PR TITLE
adding alert component

### PR DESCRIPTION
**Pull request by DeMe0**
2PR with Nick G. Alert v0

### CODE REVIEW

File: src/components/Alert/Alert.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import ColorScale from 'color-scales';
+
+import { StyledCloseButton } from './Alert.styles';
+import { AlertProps } from './types';
+
+import Container from '../Container';
+import { Paragraph } from '../Typography';
+import '../../index.css';
+
+// remove comments before pushing to master you idiot
```
Comment (nicholasgalante1997) [inline src/components/Alert/Alert.tsx]: remove this comment :)

File: src/components/Alert/Alert.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+export const DarkAlertVariantDict = {
+  info: '#08324A',
+  error: '#500202',
+  warning: '#503502',
+  success: '0B471E'
+};
```
Comment (nicholasgalante1997) [inline src/components/Alert/Alert.tsx]: add as const here, same as you did with the variant dict below

File: src/components/Alert/Alert.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+export const Alert = (props: AlertProps) => {
+  const {
+    className = 'heller-alert',
+    id = 'heller-alert-id',
+    variant = 'info',
+    title,
+    isVisible,
+    onClose,
+    customStyles,
+    maxLines = 2
```
Comment (nicholasgalante1997) [inline src/components/Alert/Alert.tsx]: we can get rid of the maxLines prop now that we've elected not to ellipsis text

File: src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  const colorPalette = new ColorScale(0, 100, [
+    AlertVariantDict[variant],
+    DarkAlertVariantDict[variant]
+  ]);
+
+  return (
+    //Parent row container
+    <Container
+      asGridParent
+      width="fit-content"
```
Comment (nicholasgalante1997) [inline src/components/Alert/Alert.tsx]: we can remove this as its never getting applied anywhere downstream

File: src/components/Alert/Alert.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      padding="0px"
+      customStyles={{
+        ...customStyles,
+        flexWrap: 'nowrap',
+        display: isVisible ? 'flex' : 'none'
+      }}
+      className={className}
+      id={id}
+    >
+      {/*10 columns of one single row  */}
```
Comment (nicholasgalante1997) [inline src/components/Alert/Alert.tsx]: we changed this to 11 cols, see colSpan below

File: src/components/Alert/types.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { WithCustomStyles } from '../../@types';
+
+export interface NativeAlertProps {
+  className?: string;
+  id?: string;
+  variant?: 'info' | 'warning' | 'error' | 'success'; //string union type
+  title: string;
+  isVisible: boolean;
+  onClose: () => void;
+  maxLines?: number;
```
Comment (nicholasgalante1997) [inline src/components/Alert/types.ts]: we can get rid of this

Comment (nicholasgalante1997) [reply src/components/Alert/types.ts]: the maxLines attribute specifically

File: src/components/Alert/Alert.test.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    expect(text).toHaveStyle({
+      color: successColorPalette.getColor(70).toHexString()
+    });
+
+    expect(buttonContainer).toHaveStyle({
+      color: successColorPalette.getColor(60).toHexString()
+    });
+  });
+
+  test('tests onClose changes isVisible property', () => {
```
Comment (nicholasgalante1997) [inline src/components/Alert/Alert.test.tsx:140]: In the future (ie not for this commit) I would like to try to add some additional "expect()" calls to this test case, Ill tag some resources later

want to be testing "hasBeenClicked(1)" for a supplied mockFn as well as the change in view state

https://jestjs.io/docs/expect#tohavebeencalled

File: src/components/Alert/Alert.tsx (hunk 4 above)
Comment (nicholasgalante1997) [reply src/components/Alert/Alert.tsx]: width attribute specifically

**Comment by DeMe0** [comment]
Praise the lord

**Pull request merged**